modules/instrument_list/templates/instrument_list_controlpanel.tpl

```diff
@@ -90,3 +98,23 @@
 			{/if}
 		</li>
 	</ul>
+
+<script>
+	function setStageAndUpdate(stageLabel) {
+		// Make an AJAX request to the endpoint
+		var xhr = new XMLHttpRequest();
```

Comment:
1. Why is this using an XMLHttpRequest AJAX request instead of the javascript fetch api?
2. Shouldn't the endpoint be returning a 303 response code to redirect? Why is there javascript needed at all?

Reply:
I had tried without using javascript and with just a redirect, however I couldn't get the redirect to do anything. I tried to look at other Loris code and came up with this:
```php
            $factory = \NDB_Factory::singleton();
            $url =
                $factory->settings()->getBaseURL()
                . '/' . $_REQUEST['lorispath']
                . '?candID=' . $_REQUEST['candID']
                . '&sessionID=' . $_REQUEST['sessionID'];
            return (new \LORIS\Http\Response())
                ->withStatus(303)
                ->withHeader("Location", $url);
            return;
```
I added this to `/instrument_list/php/instrument_list_controlpanel.class.inc` under the setStageUpdate section, the url is correct however the 303 response does not show up anywhere in my Network tab. I know that this code runs as when I error_log it shows up, but the 303 status seems to be ignored. 

I used XMLHttpRequest as I had seen it in Loris code before, but I can swap it out to the javascript fetch if you would like - if the redirect can't be fixed.

Reply:
The instrument_list_controlpanel doesn't return a PSR7 response to be returned. I just looked at the code I see how it would be complicated to add one since it's added there by the middleware in a round-about way. I guess javascript is fine, but yes, you should use the fetch API. XMLHttpRequest is a very old standard that should only be used if you need to have a progress bar showing the status of the request.